index.js

```diff
@@ -1,51 +1,70 @@
 'use strict';
 
-var path      = require('path');
-var fs        = require('fs');
-var commands  = require('./lib/commands');
+var path = require('path');
+var fs = require('fs');
+var commands = require('./lib/commands');
 var postBuild = require('./lib/tasks/post-build');
-var defaults  = require('lodash').defaults;
-var chalk     = require('chalk');
+var defaults = require('lodash').defaults;
+var chalk = require('chalk');
+
 
 module.exports = {
   name: 'ember-cli-cordova',
 
-  contentFor: function(type) {
-    if(type === 'body') {
-      return '<script src="cordova.js"></script>';
-    }
-  },
-
-  includedCommands: function() {
-    return commands;
+  _isTargetCordova: function () {
+    return !process.env.EMBER_CLI_CORDOVA ||
+      ['0', 'off', 'false', 'no'].indexOf(process.env.EMBER_CLI_CORDOVA.toLowerCase()) === -1;
   },
 
-  cdvConfig: function() {
-    var config = this.project.config('development');
-
-    if (config.environment !== 'test' && config.locationType !== 'hash') {
-      throw new Error('ember-cli-cordova: You must specify the locationType as \'hash\' in your environment.js');
+  config: function (env, config) {
+    var conf = {isCordovaBuild: this._isTargetCordova()};
+    if (conf.isCordovaBuild && env !== 'test') {
+      if (config.locationType && config.locationType !== 'hash') {
+        throw new Error('ember-cli-cordova: You must specify the locationType as \'hash\' in your environment.js or rename it to defaultLocationType.');
+      }
+      conf.locationType = 'hash';
     }
-
-    var cdvConfig = defaults(config.cordova || {}, {
+    else if (!conf.locationType) {
+      conf.locationType = config.defaultLocationType || 'auto';
+    }
+    conf.cordova = defaults(config.cordova || {}, {
       liveReload: {
-        enabled: false,
+        enabled:  false,
         platform: 'ios'
       }
     });
+    return conf;
+  },
 
-    return cdvConfig;
+  contentFor: function (type) {
+    if (this._isTargetCordova() && type === 'body') {
+      return '<script src="cordova.js"></script>';
+    }
+  },
+
+  includedCommands: function () {
+    return commands;
   },
 
-  postBuild: function() {
-    return postBuild(this.project, this.cdvConfig())();
+  cdvConfig: function () {
+    return this.project.config(process.env.EMBER_ENV || 'development').cordova;
   },
 
-  treeForPublic: function(tree) {
+  postBuild: function () {
+    if (this._isTargetCordova()) {
+      return postBuild(this.project, this.cdvConfig())();
+    }
+    else {
+      return function () {
```

Comment:
I don't think you need this `else`. It shouldn't call this function if it's `undefined`